var/spack/repos/builtin/packages/amrvis/package.py

```diff
@@ -0,0 +1,188 @@
+            'DIM = {0}'.format(spec.variants['dims'].value)
+        )
+        makefile.filter(
+            r'^PROFILE\s*=.*',
+            'PROFILE = FALSE'
+        )
+        makefile.filter(
+            r'^TRACE_PROFILE\s*=.*',
+            'TRACE_PROFILE = FALSE'
+        )
+        makefile.filter(
+            r'^COMM_PROFILE\s*=.*',
+            'COMM_PROFILE = FALSE'
+        )
+        makefile.filter(
+            r'^COMP\s*=.*',
+            'COMP = {0}'.format(self.compiler.name)
+        )
+        makefile.filter(
+            r'^DEBUG\s*=.*',
+            'DEBUG = {0}'.format(spec.variants['debug'].value).upper()
+        )
+        makefile.filter(
+            r'^USE_ARRAYVIEW\s*=.*',
+            'USE_ARRAY_VIEW = FALSE'
+        )
+        makefile.filter(
+            r'^USE_MPI\s*=.*',
+            'USE_MPI = {0}'.format(spec.variants['mpi'].value).upper()
+        )
+        makefile.filter(
+            r'^USE_CXX11\s*=.*',
+            'USE_CXX11 = TRUE'
+        )
+        makefile.filter(
+            r'^USE_VOLRENDER\s*=.*',
+            'USE_VOLRENDER = FALSE'
+        )
+        makefile.filter(
+            r'^USE_PARALLELVOLRENDER\s*=.*',
+            'USE_PARALLELVOLRENDER = FALSE'
+        )
+        makefile.filter(
+            r'^USE_PROFPARSER\s*=.*',
+            'USE_PROFPARSER = FALSE'
+        )
+
+        # A bit risky here deleting all /usr and /opt X
+        # library default search paths in makefile
+        makefile.filter(
+            r'^.*\b(usr|opt)\b.*$',
+            '# Spack removed INCLUDE_LOCATIONS and LIBRARY_LOCATIONS'
+        )
+
+        # Read GNUmakefile into array
+        with open('GNUmakefile', 'r') as file:
+            contents = file.readlines()
+
+        # Edit GNUmakefile includes and libraries to point to Spack
+        # dependencies.
+        # The safest bet is to put the LIBRARY_LOCATIONS and
+        # INCLUDE_LOCATIONS at the beginning of the makefile.
+        line_offset = 0
+        count = 0
+        for lib in ['libsm', 'libice', 'libxpm', 'libx11',
+                    'libxt', 'libxext', 'motif']:
+            contents.insert(
+                line_offset + count,
+                'LIBRARY_LOCATIONS += {0}\n'.format(spec[lib].prefix.lib)
+            )
+            contents.insert(
+                line_offset + count + 1,
+                'INCLUDE_LOCATIONS += {0}\n'.format(spec[lib].prefix.include)
+            )
+            count = count + 1
+
+        # Write GNUmakefile
+        with open('GNUmakefile', 'w') as file:
+            file.writelines(contents)
+
+    def setup_environment(self, build_env, run_env):
+        # Help force Amrvis to not pick up random system compilers
+        if '+mpi' in self.spec:
+            build_env.set('MPI_HOME', self.spec['mpi'].prefix)
+            env['CC'] = spec['mpi'].mpicc
+            env['CXX'] = spec['mpi'].mpicxx
+            env['F77'] = spec['mpi'].mpif77
+            env['FC'] = spec['mpi'].mpifc
```

Comment:
You should use `build_env.set` for these. Also, every other package calls it `spack_env`, not `build_env`. I agree that the name is kind of confusing, but we should keep things consistent. If you want to convert every package to `build_env` in another PR, I would support that decision.